src/resources.rs

```diff
@@ -194,151 +194,88 @@ pub fn lookup(rid: ResourceId) -> Option<Resource> {
   table.get(&rid).map(|_| Resource { rid })
 }
 
-type EagerRead<R, T> =
+pub type EagerRead<R, T> =
   Either<tokio_io::io::Read<R, T>, FutureResult<(R, T, usize), std::io::Error>>;
 
-#[cfg(windows)]
+pub type EagerWrite<R, T> =
+  Either<tokio_write::Write<R, T>, FutureResult<(R, T, usize), std::io::Error>>;
+
+pub type EagerAccept = Either<
+  tokio_util::Accept,
+  FutureResult<(tokio::net::TcpStream, std::net::SocketAddr), std::io::Error>,
+>;
+
+#[cfg(not(unix))]
 #[allow(unused_mut)]
-pub fn eager_read<T>(resource: Resource, mut buf: T) -> EagerRead<Resource, T>
-where
-  T: AsMut<[u8]>,
-{
+pub fn eager_read<T: AsMut<[u8]>>(
+  resource: Resource,
+  mut buf: T,
+) -> EagerRead<Resource, T> {
   Either::A(tokio_io::io::read(resource, buf)).into()
 }
 
+#[cfg(not(unix))]
+pub fn eager_write<T: AsRef<[u8]>>(
```

Comment:
Just call this write() 

and similarly 

s/eager_accept/accept/
s/eager_write/write/

Reply:
I went back and forth on this, and decided to rename it back to eager_write to match the return type EagerWrite, which I kept because in the same file there is also mention of the AsyncWrite (which is really NonBlockginWrite) and Write (which is really AsyncWrite).